From c0f664218e4e5462d1282b015439c02ad9e9993d Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Thu, 9 Nov 2017 19:48:27 +0100
Subject: [PATCH] HEAD-2375 binder instances created after HN setup don't work
 with sapi-url being a domain Reviewed by: Trent Mick <trentm@gmail.com>
 Approved by: Trent Mick <trentm@gmail.com>

---
 scripts/headnode.sh | 10 ----------
 scripts/sdc-init.js |  2 +-
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/scripts/headnode.sh b/scripts/headnode.sh
index 62589f7d..7e601941 100755
--- a/scripts/headnode.sh
+++ b/scripts/headnode.sh
@@ -850,16 +850,6 @@ EOF
 {"set_customer_metadata": {"sapi-url": "http://${CONFIG_sapi_domain}"}}
 EOF
 
-    # Update binder0 zone sapi-url
-    vmadm update $(vmadm lookup -1 tags.smartdc_role=binder) <<EOF
-{"set_customer_metadata": {"sapi-url": "http://${CONFIG_sapi_domain}"}}
-EOF
-    # Update binder service metadata.sapi-url
-    binder_svc_uuid=$(/opt/smartdc/bin/sdc-sapi \
-                      /services?name=binder|json -H 0.uuid)
-    /opt/smartdc/bin/sapiadm update ${binder_svc_uuid} \
-      metadata.sapi-url=http://${CONFIG_sapi_domain}
-
     setup_state_add "sdczones_created"
 fi
 
diff --git a/scripts/sdc-init.js b/scripts/sdc-init.js
index 8cec9db5..d4bf00b9 100755
--- a/scripts/sdc-init.js
+++ b/scripts/sdc-init.js
@@ -462,7 +462,7 @@ function filterServices(serviceList, cb) {
             // We'll use IP at first pass, since sapi service is either not
             // running when we create these zones or not yet registered into
             // binder. Then, we'll update at the end of the setup process.
-            if (service === 'binder' ||
+            if (service === 'sapi' ||
                 service === 'assets' ||
                 service === 'binder') {
                 extras.metadata['sapi-url'] =
-- 
2.21.0

